homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,172 @@
+)
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import ATTR_COUNTY, ATTRIBUTION, CONF_DISTRICT_NAME, CONF_DISTRICTS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    "count": "mdi:virus",
+    "deaths": "mdi:cross",
+    "recovered": "mdi:bottle-tonic-plus-outline",
+    "week_incidence": "mdi:clipboard-pulse",
+    "cases_per100k": "mdi:home-group",
+    "new_cases": "mdi:shield-bug",
+    "new_deaths": "mdi:shield-cross",
+    "new_recovered": "mdi:shield-sync",
+}
+
+DISTRICT_SENSORS = {
+    "hospitalization_cases_baby": "mdi:baby-face-outline",
+    "hospitalization_incidence_baby": "mdi:baby-face",
+    "hospitalization_cases_children": "mdi:account-child-outline",
+    "hospitalization_incidence_children": "mdi:account-child",
+    "hospitalization_cases_teen": "mdi:face-woman",
+    "hospitalization_incidence_teen": "mdi:face-woman-outline",
+    "hospitalization_cases_grown": "mdi:face-man",
+    "hospitalization_incidence_grown": "mdi:face-man-outline",
+    "hospitalization_cases_senior": "mdi:account-cowboy-hat-outline",
+    "hospitalization_incidence_senior": "mdi:account-cowboy-hat",
+    "hospitalization_cases_old": "mdi:human-white-cane",
+    "hospitalization_incidence_old": "mdi:human-cane",
+}
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+    coordinator = RkiCovidDataUpdateCoordinator(hass)
+    await coordinator.async_config_entry_first_refresh()
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    districts = config[CONF_DISTRICTS]
+
+    # district sensors
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district[CONF_DISTRICT_NAME], info_type)
+        for info_type in SENSORS
+        for district in districts
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+    # sensors on state level (additional sensors)
+    for district in districts:
+        if district.startswith("BL"):
+            district_sensors = [
+                RKICovidNumbersSensor(coordinator, district, info_type)
+                for info_type in DISTRICT_SENSORS
+            ]
+            async_add_entities(district_sensors, update_before_add=True)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Create sensors from a config entry in the integrations UI."""
+    _LOGGER.debug("create sensor from config entry %s", config_entry.data)
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
```

Comment:
Not needed as it would not come this far if coordinator or data not available

Reply:
done